Docker image unable to run migrations

The docker image doesn't work. At the running migrations step, it fails as it is unable to connect to the postgres docker container.

```
==> Checking for blockscout image blockscout_prod
==> Image exist. Using blockscout_prod
==> Checking PostrgeSQL container
==> Starting PostgreSQL container
postgres
==> Running migrations


16:46:09.076 [error] GenServer #PID<0.380.0> terminating
** (DBConnection.ConnectionError) tcp connect (localhost:5432): connection refused - :econnrefused
    (db_connection 2.2.2) lib/db_connection/connection.ex:87: DBConnection.Connection.connect/2
    (connection 1.0.4) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 3.12.1) proc_lib.erl:249: :proc_lib.init_p_do_apply/3
Last message: nil
State: Postgrex.Protocol
** (Mix) The database for Explorer.Repo couldn't be created: killed
make: *** [migrate] Error 1
```

Looking at the logs of the postgres container, we can see that there is a configuration error.

```
Error: Database is uninitialized and superuser password is not specified.
       You must specify POSTGRES_PASSWORD to a non-empty value for the
       superuser. For example, "-e POSTGRES_PASSWORD=password" on "docker run".

       You may also use "POSTGRES_HOST_AUTH_METHOD=trust" to allow all
       connections without a password. This is *not* recommended.

       See PostgreSQL documentation about "trust":
       https://www.postgresql.org/docs/current/auth-trust.html
```

Seems like Postgres 12 doesn't set it to trust mode by default and requires a password to be set. I think we can set the auth method to trust to mirror the old behaviour.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 
 
 ### Fixes
+- [#3437](https://github.com/poanetwork/blockscout/pull/3437) - Fix Postgres Docker container
 - [#3428](https://github.com/poanetwork/blockscout/pull/3428) - Fix address tokens search
 - [#3424](https://github.com/poanetwork/blockscout/pull/3424) - Fix display of long NFT IDs
 - [#3422](https://github.com/poanetwork/blockscout/pull/3422) - Fix contract reader: tuple type

--- a/docker/Makefile
+++ b/docker/Makefile
@@ -306,6 +306,7 @@ else
 	@docker run -d --name $(PG_CONTAINER_NAME) \
 					-e POSTGRES_PASSWORD="" \
 					-e POSTGRES_USER="postgres" \
+					-e POSTGRES_HOST_AUTH_METHOD="trust" \
 					-p 5432:5432 \
 					$(PG_CONTAINER_IMAGE)
 	@sleep 1

